lib/material_radio/material_radio.dart

```diff
@@ -120,48 +102,38 @@ class MaterialRadioComponent extends RootFocusable
 
   /// Whether the radio should be preselected.
   @Input()
-  set checked(bool newValue) {
-    if (_checked == newValue) return;
-    // TODO(google): A video test fails without this MarkForCheck, but
-    // it shouldn't be needed for @Inputs.
+  set checked(bool isChecked) {
+    if (_checked == isChecked) return;
+    _checked = isChecked;
     _changeDetector.markForCheck();
 
-    _icon = newValue ? checkedIcon : uncheckedIcon;
-
     if (_group != null) {
-      if (newValue) {
+      if (isChecked) {
         _group.componentSelection.select(this);
       } else {
         _group.componentSelection.deselect(this);
       }
     }
-
-    _checked = newValue;
-    _syncAriaChecked();
     _onChecked.add(_checked);
   }
 
+  @HostBinding('attr.aria-checked')
   bool get checked => _checked;
   bool _checked = false;
 
   /// Current icon, depends on the state of [checked].
-  Icon get icon => _icon;
-  Icon _icon = uncheckedIcon;
+  Icon get icon => _checked ? checkedIcon : uncheckedIcon;
 
   /// Current tab index, depends on state of [disabled] and selection status
   /// if in group.
-  String get tabIndex => '$_tabIndex';
-  int _tabIndex = 0;
-  int _enabledTabIndex = 0;
+  @HostBinding('attr.tabindex')
+  int get tabIndex => disabled ? -1 : _enabledTabIndex;
```

Comment:
In `lib/material_checkbox/material_checkbox.dart`, we have essentially:

```dart
  @HostBinding('attr.tabindex')
   String get tabIndex => disabled ? '-1' : _defaultTabIndex;
```

Should the getter return type be `String` here too?

(Should the class extend `HasTabIndex`?)

Reply:
I know there is cleanup/refactor going on. I'll ask the author.